hooks/usePaginatedSocialList.ts

```
@@ -0,0 +1,102 @@
+ * mutations (e.g. accepting two requests back to back) can't resolve out of
+ * order and leave stale data in state.
+ */
+export function usePaginatedSocialList<T>({ accessToken, enabled, baseUrl, onData }: Options<T>) {
+  const [items, setItems] = useState<T[]>([]);
+  const [hasMore, setHasMore] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [loadingMore, setLoadingMore] = useState(false);
+  const [error, setError] = useState(false);
+
+  const itemsRef = useRef(items);
+  itemsRef.current = items;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/hooks/usePaginatedSocialList.ts:40:3
  38 |
  39 |   const itemsRef = useRef(items);
> 40 |   itemsRef.current = items;
     |   ^^^^^^^^^^^^^^^^ Cannot update ref during render
  41 |   const abortRef = useRef<AbortController | null>(null);
  42 |   const onDataRef = useRef(onData);
  43 |   onDataRef.current = onData;

(react-hooks/refs)